### PR TITLE
Add release notes for chefdk 3.4.

### PR DESCRIPTION
**Pull request by jonsmorrow**
Signed-off-by: Jon Morrow <jmorrow@chef.io>

### Description

☝️ 

### Check List

- [X] Spell Check
- [X] Local build
- [X] Examine the local build
- [ ] All tests pass

### CODE REVIEW

File: chef_master/source/release_notes_chefdk.rst [hunk 1]
```diff
@@ -5,6 +5,61 @@ Release Notes: Chef Development Kit 0.19 - 3.3
...
+
+  macOS 10.14 (Mojave) is now fully tested and packages are available on the `downloads <https://downloads.chef.io/chefdk/3.4.38>`__ page.
+
+* **Updated Tooling**
+
+  **Fauxhai**
+
+    Foodcritic has been updated from 14.1 to 14.3. This updates the metadata that ships with Foodcritic to provide the latest Chef 13.11 and 14.5 metadata, while removing metadata from older Chef releases. This update also removes the FC121 rule, which was causing confusion with community cookbook authors. This rule will be added back when Chef 13 goes EOL in April 2019.
+
+  **inSpec 3**
```
Comment (tas50) [inline chef_master/source/release_notes_chefdk.rst]: The i is capitalized

File: chef_master/source/release_notes_chefdk.rst [hunk 2]
```diff
@@ -5,6 +5,61 @@ Release Notes: Chef Development Kit 0.19 - 3.3
...
+
+    This release updates Inspec from 2.2.112 to 3.0.12. This is a major milestone and includes the plugin system, global attributes, enhanced skip messaging, and more. Please head over to the `inSpec <https://www.inspec.io/>`__ site for a full rundown.
```
Comment (tas50) [inline chef_master/source/release_notes_chefdk.rst]: The S in inspec needs to be capitalized

File: chef_master/source/release_notes_chefdk.rst [hunk 3]
```diff
@@ -1,10 +1,65 @@
...
+
+* **Platform Additions**
+
+  macOS 10.14 (Mojave) is now fully tested and packages are available on the `downloads <https://downloads.chef.io/chefdk/3.4.38>`__ page.
+
+* **Updated Tooling**
+
+  **Fauxhai**
+
+    Foodcritic has been updated from 14.1 to 14.3. This updates the metadata that ships with Foodcritic to provide the latest Chef 13.11 and 14.5 metadata, while removing metadata from older Chef releases. This update also removes the FC121 rule, which was causing confusion with community cookbook authors. This rule will be added back when Chef 13 goes EOL in April 2019.
```
Comment (marcparadise) [inline chef_master/source/release_notes_chefdk.rst:30]: > oodcritic has been updated from 14.1 to 14.3. This updates the metadata that ships with Foodcritic to provide the latest Chef 13.11 and 14.5 metadata, while removing metadata from older Chef releases

Suggest: Foodcritic has been updated from 14.1 to 14.3. This brings its metadata up to date with Chef 13.11 and 14.5, and removes older Chef metadata.

File: chef_master/source/release_notes_chefdk.rst [hunk 4]
```diff
@@ -1,10 +1,65 @@
...
+
+    The Kitchen DigitalOcean driver now supports FreeBSD 10.4 and 11.2 in Kitchen configs.
+
+  **Kitchen EC2**
+
+    Kitchen EC2 has been updated to better support Windows systems. The auto-generated security group will now include support for RDP and the log directory will alway be created.
+
+  **Kitchen Google**
+
+    Kitchen Google now includes support for adding labels to instances with a new `labels` config that accepts labels as a hash.
```
Comment (marcparadise) [inline chef_master/source/release_notes_chefdk.rst:50]: Would it be good to give an in-context example config?

File: chef_master/source/release_notes_chefdk.rst [hunk 5]
```diff
@@ -1,10 +1,65 @@
...
+
+  **Knife Windows**
+
+    Knife Windows has improved Windows detection support to identify Windows 2012r2, 2016, and 10. Additionally when bootstrapping nodes, there is now support for using the client.d directories.
+
+* **Security Updates**
```
Comment (marcparadise) [inline chef_master/source/release_notes_chefdk.rst:56]: I think we should include Security at or near the top, ahead of product-level changes. CVEs can be the most critical consideration when deciding whether or not to upgrade.

**Comment by kagarmoe** [comment]
Added through different PR

**Pull request closed without merging**